milli/src/update/settings.rs

```diff
@@ -210,6 +220,31 @@ impl<'a, 't, 'u, 'i> Settings<'a, 't, 'u, 'i> {
         Ok(true)
     }
 
+    fn update_stop_words(&mut self) -> anyhow::Result<bool> {
+        match self.stop_words {
+            Some(Some(ref stop_words)) => {
+                let current = self.index.stop_words(self.wtxn)?;
+                // since we can't compare a BTreeSet with an FST we are going to convert the
+                // BTreeSet to an FST and then compare bytes per bytes the two FSTs.
+                let fst = fst::Set::from_iter(&*stop_words)?;
+
+                // Does the new FST differ from the previous one?
+                if current.map_or(true, |current| current.as_fst().as_bytes() != fst.as_fst().as_bytes()) {
+                    // we want to re-create our FST.
+                    self.index.put_stop_words(self.wtxn, &fst)?;
+                    Ok(true)
+                } else {
+                    Ok(false)
+                }
+            }
+            Some(None) => {
+                self.index.delete_stop_words(self.wtxn)?;
+                Ok(true)
+            }
```

Comment:
```suggestion
            Some(None) => Ok(self.index.delete_stop_words(self.wtxn)?),
```

It is better to return `true` only if there was a previous list of `stop_words`: if we were really modifying the database.

https://docs.rs/heed/0.10.6/heed/struct.Database.html#method.delete